app/helpers/bex_helper.rb

```diff
@@ -1,20 +1,23 @@
 module BexHelper
   # Agenda JAP
-  def ten_next_fridays
-    fridays = [Date.today.next_occurring(:friday)]
-
-    9.times { fridays << (fridays[-1] + 1.week) }
-
-    fridays
+  def ten_next_days_with_slots(appointment_type, organization)
+    Slot.future
+        .in_organization(organization)
+        .where(appointment_type: appointment_type)
+        .pluck(:date)
+        .uniq
+        .sort
+        .first(10)
   end
 
-  def available_agendas
-    ['Cabinet 1', 'Cabinet 2', 'Cabinet 3', 'Cabinet 4',
-     'Cabinet 5', 'Cabinet 6', 'Cabinet 7']
+  def first_day_with_slots(appointment_type, organization)
+    ten_next_days_with_slots(appointment_type, organization).first
   end
 
-  def bex_jap_available_hours
-    %w[09h30 10h00 10h30 11h00 11h30]
+  def available_slots_hours(date, agenda)
```

Comment:
Pareil, je serais pas choqué que ce soit une méthode d'instance. On ne l'utilise pas que comme de l'affichage mais pour construire un enum sur lequel on itère